### ISSUE TITLE
Websockets not allowing connections for some clients

### ISSUE THREAD

**Issue by samaklis** [eclipse/mosquitto#1721]
Hi, I am running Mosquitto 1.4.10 on a Raspberrypi. I have had my configuration working seamlessly for quite a long time with SSL over websockets. I have MQTT Paho java clients hitting the Mosquitto setup, and also python MQTT clients.  

Recently I noticed that sometimes I would get a denied connection and some odd message about a certificate being expired (which is not the case) so I decided to enable some logging on the websockets. 

At this point in time the only device that has a problem connecting to the MQTT server is my iPhone running Pythonista using the Paho Python client. The same exact code on my iPad works fine. This was working fine up to a couple of weeks or so ago so I am not sure if there was any Mosquitto updates that introduced this issue. Also the fact that it is happening only on one client is kind of perplexing.

I am attaching the Mosquitto log, showing the "rejection" for my iPhone client and then the successful connection from my iPad.

Any insight as to if this is a possible bug or some other issue would be appreciated.
[mosquitto.log](https://github.com/eclipse/mosquitto/files/4751526/mosquitto.log)

**Comment by ralight** [comment]
I wonder if this might be an update to your iPhone making it stricter on what certificates it accepts. I've just had to update the broker certificates on test.mosquitto.org for the same reason.

There weren't any updates to the Mosquitto version you have - that version is nearly 4 years old now. Upgrading to a new version of raspbian would get you a more recent version.

**Comment by samaklis** [comment]
Thanks for taking the time to get back to me. When you say that you "updated" the certificates what do you mean? Mine are done through third party (namecheap). I had them renewed last year. Are you saying there is a way to somehow request a newer type of SSL?

Thanks

**Comment by ralight** [comment]
These only had 1024 bit rsa keys and sha1 hashes. If you renewed your certificate last year I can't imagine that would be the problem.

Something you could try to debug this is use the `openssl` tool. Run `openssl s_client -connect <host>:<port>` then look for the `Verify return code` line to see what it says.

**Comment by samaklis** [comment]
I have narrowed it down to the version of openssl that comes packaged with the Pythonista application comes with. (OpenSSL 1.0.2f 28 Jan 2016)
It looks like after you upgrade iOS to anything over 14.5 (15.1, etc.) it starts to break things. 
Later versions (I tested with: OpenSSL 1.1.1f 31 Mar 2020) work as expected.

I have submitted a bug with their dev team to have a look and upgrade the included version of openssl.

Thanks for all your help
